File: Web/src/api/xzmenu/xzAdv.ts
import {useBaseApi} from '/@/api/base';

// 系统广告接口服务
export const useXzAdvApi = () => {
	const baseApi = useBaseApi("xzAdv");
	return {
		// 分页查询系统广告
		page: baseApi.page,
		// 查看系统广告详细
		detail: baseApi.detail,
		// 新增系统广告
		add: baseApi.add,
		// 更新系统广告
		update: baseApi.update,
		// 删除系统广告
		delete: baseApi.delete,
		// 批量删除系统广告
		batchDelete: baseApi.batchDelete,
		// 导出系统广告数据
		exportData: baseApi.exportData,
		// 导入系统广告数据
		importData: baseApi.importData,
		// 下载系统广告数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 系统广告实体
export interface XzAdv {
	// 主键Id
	id: number;
	// 广告名称
	name: string;
	// 广告图片
	img: string;
	// 广告位置
	postion: number;
	// 开始时间
	stime: string;
	// 结束时间
	etime: string;
	// 排序
	sortcode: number;
	// 跳转地址
	url: string;
	// 0：启用，1：不显示
	isenable: number;
	// 点击次数
	click: number;
	// 时间
	createtime: string;
}

File: Admin.NET/Admin.NET.Application/Entity/XzAdv.cs
// Admin.NET 项目的版权、商标、专利和其他相关权利均受相应法律法规的保护。使用本项目应遵守相关法律法规和许可证的要求。
//
// 本项目主要遵循 MIT 许可证和 Apache 许可证（版本 2.0）进行分发和使用。许可证位于源代码树根目录中的 LICENSE-MIT 和 LICENSE-APACHE 文件。
//
// 不得利用本项目从事危害国家安全、扰乱社会秩序、侵犯他人合法权益等法律法规禁止的活动！任何基于本项目二次开发而产生的一切法律纠纷和责任，我们不承担任何责任！

using Admin.NET.Core;
using SqlSugar;
namespace Admin.NET.Application.Entity;

/// <summary>
/// 系统广告
/// </summary>
[Tenant("1300000000001")]
[SugarTable("xz_adv", "系统广告")]
public partial class XzAdv : EntityBaseId
{
    /// <summary>
    /// 广告名称
    /// </summary>
    [SugarColumn(ColumnName = "name", ColumnDescription = "广告名称", Length = 100)]
    public virtual string? name { get; set; }
    
    /// <summary>
    /// 广告图片
    /// </summary>
    [SugarColumn(ColumnName = "img", ColumnDescription = "广告图片", Length = 300)]
    public virtual string? img { get; set; }
    
    /// <summary>
    /// 广告位置
    /// </summary>
    [SugarColumn(ColumnName = "postion", ColumnDescription = "广告位置")]
    public virtual int? postion { get; set; }
    
    /// <summary>
    /// 开始时间
    /// </summary>
    [SugarColumn(ColumnName = "stime", ColumnDescription = "开始时间")]
    public virtual DateTime? stime { get; set; }
    
    /// <summary>
    /// 结束时间
    /// </summary>
    [SugarColumn(ColumnName = "etime", ColumnDescription = "结束时间")]
    public virtual DateTime? etime { get; set; }
    
    /// <summary>
    /// 排序
    /// </summary>
    [SugarColumn(ColumnName = "sortcode", ColumnDescription = "排序")]
    public virtual int? sortcode { get; set; }
    
    /// <summary>
    /// 跳转地址
    /// </summary>
    [SugarColumn(ColumnName = "url", ColumnDescription = "跳转地址", Length = 300)]
    public virtual string? url { get; set; }
    
    /// <summary>
    /// 0：启用，1：不显示
    /// </summary>
    [SugarColumn(ColumnName = "isenable", ColumnDescription = "0：启用，1：不显示")]
    public virtual int? isenable { get; set; }
    
    /// <summary>
    /// 点击次数
    /// </summary>
    [SugarColumn(ColumnName = "click", ColumnDescription = "点击次数")]
    public virtual int? click { get; set; }
    
    /// <summary>
    /// 时间
    /// </summary>
    [SugarColumn(ColumnName = "createtime", ColumnDescription = "时间")]
    public virtual DateTime? createtime { get; set; }
    
}
